### ISSUE TITLE
Implementing geodesic distance for numerical optimization

### ISSUE THREAD

**Issue by nquesada** [polyquantique/Unitary-Decomp#3]
In https://arxiv.org/abs/2510.19397 it is argued that the "correct" cost function to do gradient descent is

<img width="397" height="105" alt="Image" src="https://github.com/user-attachments/assets/89329ca1-33d3-4ec7-9e00-9e62b7288ff9" />

It would be nice to implement this cost function with gradients in jax. Note that jax does not have a differentiable matrix log function. However, one can go very far using their `schur` decomposition.

**Comment by nquesada** [comment]
this could be an easy addition: `jnp.sum(jnp.angle(jnp.linalg.eigvals(U))**2)`
